### DISCUSSION TITLE
Uploading to two objects stores with one sidecar

### DISCUSSION THREAD

**Discussion by sinkingpoint** [thanos-io/thanos#6109] [Questions & Answers]
Is there a way to have a thanos sidecar upload to two seperate buckets? I want to replicate my Prometheus blocks to two different object stores (internal, and a cloud for redundancy) but it seems like a sidecar only allows uploading to one. Running two sidecars doesn't work because the thanos.shipper.json prevents the second from uploading blocks that have already been uploaded by the first.

**Comment by matej-g** [comment]
Hey @sinkingpoint,
Not really sure about the 2 sidecars use case. Is there a reason why replication between the object stores is not viable?

**Reply by sinkingpoint** [reply]
Mostly from an operational standpoint. It's much easier to run two of the same thing, instead of maintaining two different things

**Reply by yeya24** [reply]
So if we make `thanos.shipper.json` configurable per shipper then it should work for you?

**Reply by sinkingpoint** [reply]
Yep. In fact that's what I've patched our internal version to allow already and it works well
